packages/effect/test/Pool.test.ts

```diff
@@ -202,6 +217,70 @@ describe("Pool", () => {
       expect(max).toBe(15)
     }))
 
+  it.scoped("max pool size with concurrency: 3", () =>
+    Effect.gen(function*($) {
```

Comment:
I see us removing the adapter in some tests but keeping it elsewhere. I suppose it doesn't really matter for this PR, but perhaps we should consider stripping it out of the tests entirely at some point.

Reply:
The ts inference issue keeps popping up, so I think we should wait until 5.5 lands.